ocaml/xenopsd/scripts/swtpm-wrapper

```diff
@@ -85,38 +83,84 @@ def prepare_exec():
     sys.stdout.flush()
     sys.stderr.flush()
 
+def make_socket(fname):
+
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.bind(fname)
+    sock.listen(1)
+
+    return sock
+
+def check_state_needs_init(fname):
+
+    if not os.path.exists(fname):
+        return True
+
+    mode = os.stat(fname).st_mode
+
+    if not stat.S_ISBLK(mode):
+        return False
+
+    # Check if block device has non-zero header
+    file = open(fname, "r")
+    hdr = file.read(8)
+    file.close()
+
+    if len(hdr) == 8 and hdr != "\0\0\0\0\0\0\0\0":
+        return False
+    return True
+
+
 def main(argv):
     print("Arguments: %s" % " ".join(argv[1:]))
 
-    if len(argv) < 3:
-        return 
+    if len(argv) < 4:
+        sys.exit("Not enough arguments.")
 
     domid = int(argv[1])
     tpm_dir = argv[2]
+    tpm_state = argv[3]
     tpm_path = tpm_dir
     depriv = True
 
-    n= 3
+    n = 4
     while n < len(argv):
-        if argv[n] == "-priv":
+        if argv[n] == "--priv":
             depriv = False
-            continue
+        else:
+            sys.exit("Unknown option %s\n" % argv[n])
         n += 1
 
+    uri_sep = tpm_state.find("://")
+    tpm_state_prefix = tpm_state[:uri_sep]
+    tpm_state_file = tpm_state[uri_sep+3:].lstrip('/')
+
+    if (tpm_state_prefix == "dir"):
+        tpm_uri = tpm_dir
+        tpm_file = os.path.join(tpm_dir, tpm_state_file)
```

Comment:
tpm_file has the same definiton in both branches, should be defined before to avoid repetition